libs/server-sdk/src/client_impl.cpp

```diff
@@ -125,8 +127,12 @@ static bool IsInitialized(DataSourceStatus::DataSourceState state) {
 }
 
 void ClientImpl::Identify(Context context) {
-    event_processor_->SendAsync(events::IdentifyEventParams{
-        std::chrono::system_clock::now(), std::move(context)});
+    if (events_default_.disabled) {
+        return;
+    }
+
+    event_processor_->SendAsync(
+        events_default_.factory.Identify(std::move(context)));
 }
```

Comment:
There's a slight bit of weirdness here. It'd be nice to not have to do _any_ work if events are disabled - like is done here with the early out.

This skips any copies or constructors that would otherwise be called and then subsequently discarded by the `NullEventProcessor`, if events are disabled.

But then - why not just always construct a real event processor?
- Don't waste a thread servicing the real event processor's timer
- Speeds up instantiation of the SDK probably by a tiny amount
- Ensures events are not sent, event if `EventScope.disabled` isn't checked

Could also remove this boolean altogether and just keep relying on the `NullEventProcessor`. I copied this technique from how we do it in Rust.